service/worker/archiver/activities.go

```diff
@@ -225,6 +232,53 @@ func uploadHistoryActivity(ctx context.Context, request ArchiveRequest) (err err
 		logger.Error(uploadErrorMsg, tag.ArchivalUploadFailReason(errorDetails(err)), tag.ArchivalBlobKey(indexBlobKey.String()), tag.Error(err))
 		return err
 	}
+	if runBlobIntegrityCheck {
```

Comment:
Can we move this to a separate function? The code of upload activity is getting longer and longer.

Reply:
+1 - consider factoring out into its own function

Reply:
I want to leave it as part of the same method for now. By making it its own function you have to pass two many parameters and it doesn't buy you much because the whole logic is already inside an if condition, so the reader can easily mentally skip over the section already.